src/Symfony/Component/Yaml/Inline.php

```diff
@@ -124,7 +133,13 @@ public static function dump($value, $flags = 0)
                 }
 
                 if (Yaml::DUMP_OBJECT_AS_MAP & $flags && ($value instanceof \stdClass || $value instanceof \ArrayObject)) {
-                    return self::dumpArray($value, $flags & ~Yaml::DUMP_EMPTY_ARRAY_AS_SEQUENCE);
+                    $output = array();
+
+                    foreach ($value as $key => $val) {
+                        $output[] = sprintf('%s: %s', self::dump($key, $flags), self::dump($val, $flags));
+                    }
+
+                    return sprintf('{ %s }', implode(', ', $output));
```

Comment:
Is this code change intended in this PR?

Reply:
Yes, this part is handling the dumping of objects as YAML mappings and thus allows us to type hint the `$value` argument in `dumpArray()` with `array`.